### PR TITLE
LPD-17848 Functional automation for saving and editing of DSM list sections

### PR DESCRIPTION
**Pull request by markocikos**
### References

- Parent Story: [LPD-6624 Allow admins to save the definition for list as visualization mode](https://issues.liferay.com/browse/LPD-6624)

### What is the goal of this PR?

The main purpose of this PR is to add tests for assigning fields to of DSM list sections. In addition to it, we're
- adding a helper to create DSM data with API
- tweaking DSM code for easier tests, by adding container CSS classes and setting toast as last
- fixing a flakiness issue in existing tests
- adding default sample entity labels in constants

See technical notes inline.

### What does it look like?

image

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/ListVisualizationMode.scss
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/ListVisualizationMode.scss:4]: `list-visualization-mode` is relevant for tests. `field-name` and `list-section-label` are renames I missed in https://github.com/liferay-frontend/liferay-portal/pull/3990

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx:153]: Setting toast as last allows us to do 
```js
await listVisualizationModePage.page.getByText('Success').waitFor();
```
The idea is, when the toast appears, `setListSections` is resolved and the field that we are about to check has new value. This is tiny bit scary to me, as React component needs to reload with updated state, but I didn't notice any flakiness. I am not sure if there is any cleaner approach to this.

File: modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts:22]: This is temporary. What we should do is rewrite Fields tests to use new UI. I can do this in a followup.

File: modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts:81]: @chestofwonders This test is broken for me, I get is 404 error to a broken-looking URL. Also, this test should not be in this file, `fields.spec.ts` should hold tests related to Fields. We can fix this in a separate task.

image

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/ViewsPage.ts
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/ViewsPage.ts:59]: I renamed this from `goto*` to `open*`. I am suggesting here a naming convention:
- `goto*` implies navigation from home page to the end page.
- any other navigation local in page should be named something else, `open*`, `select*` etc.

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetsPage.ts
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetsPage.ts:108]: @juanjofgliferay I believe you noticed the same issue in https://github.com/liferay-frontend/liferay-portal/pull/4002, that check for `heading` is not sufficient. You added
```js
await page.locator('.dnd-table > .dnd-thead > .dnd-tr').waitFor();
```
This does basically the same thing, but also provides a handy container reference, for easier locators. I added several of these container locators in PR, even adding CSS classes in code to enable them.

Comment (juanjofgliferay) [reply modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetsPage.ts:108]: @markocikos Yep, I come to the same conclusion. This is something we need to do at this point, otherwise we will end up using `waitFor()` across the tests

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/ViewsPage.ts
Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/pages/ViewsPage.ts:54]: In DSM code, we call this `label`, in [both UI code and persistence](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx#L747).

In general, there is a loose convention in DXP: a `name` implies the value is a unique string identifier. Examples would be `fieldName`, `fdsName`, `name` attribute of form elements etc. `id`, conversely, implies the value is a unique _number_ identifier. `label` implies that the value is just for UI, non-unique, likely localized.

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:test:relevant

File: modules/test/playwright/tests/frontend-data-set-views-web/helpers/DataSetManagerApiHelpers.ts
Comment (izaera) [reply modules/test/playwright/tests/frontend-data-set-views-web/helpers/DataSetManagerApiHelpers.ts:10]: I guess it's because `ApiHelpers` will be deleted if the headless PR gets merged one day so no one is thinking in refactoring or changing that... :shrug:

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/ViewsPage.ts
Comment (juanjofgliferay) [reply modules/test/playwright/tests/frontend-data-set-views-web/pages/ViewsPage.ts:54]: I'll apply the changes in a followup PR

**Comment by juanjofgliferay** [comment]
> ci:test:playwright

@markocikos I think that this will be useless until we've got #4002 merged

I've realized that our project is not the list of [playwright tests](https://github.com/liferay-frontend/liferay-portal/blob/master/test.properties#L923) setup.

We should consider adding a section in the test.properties file to run frontend-data-set test when executing `relevant` . See https://liferay.atlassian.net/wiki/spaces/ENGR/pages/2575566031/Using+Microsoft+Playwright+in+Liferay+CI#Additional-test-suites

File: modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts
Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts:81]: The test is [failing in CI](https://github.com/liferay-frontend/liferay-portal/pull/4002#issuecomment-1956649039), once @juanjofgliferay enabled it.

**Comment by markocikos** [comment]
> We should consider adding a section in the test.properties file to run frontend-data-set test when executing relevant

Hmm I always assumed that `relevant` is a special "smart" test, that runs tests based on modules with changes. I'm not sure if adding playwright somewhere [here](https://github.com/liferay/liferay-portal/blob/master/test.properties#L8578) would be the right way to go. I don't know how to achieve something like: when there is a change in `frontend-data-set-views` module, run `frontend-data-set-views` tests. The `relevant` suite in 

What we could do is make a new `[frontend]` suite for playwright, something like
```
    playwright.test.project[playwright-js-tomcat90-mysql57-jdk8][frontend]=\
        client-extension-web,\
        frontend-data-set-views-web
```
@dsanz what do you think?

I see Echo did [something like that](https://github.com/liferay/liferay-portal/blob/master/test.properties#L4866-L4867). How do you run just one suite? :thinking:

**Comment by juanjofgliferay** [comment]
@markocikos I also assumed that `relevant` is black magic until I read this [slack thread](https://liferay.slack.com/archives/C06A59K5D46/p1707242909686339?thread_ts=1707162744.755759&cid=C06A59K5D46) , were they expect development teams to configure its usage

**Comment by markocikos** [comment]
> expect development teams to configure its usage

Oooh I see. Yeah, IMO we should definitely do that.

**Comment by juanjofgliferay** [comment]
@markocikos In order to run only one suite, they recommend comment all playwright projects except the one you want. Also, you could use `ci : test:playwright:nocompile` to use the latest bundle. See [slack thread](https://liferay.slack.com/archives/C06A59K5D46/p1707504553246879)

**Comment by markocikos** [comment]
> comment all playwright projects except the one you want

:smile:

**Comment by markocikos** [comment]
ci:test:playwright

**Review by juanjofgliferay — APPROVED** [review]
LGTM :rocket:

**Comment by markocikos** [comment]
ci:report:2541860254

**Comment by markocikos** [comment]
No unique failures on relevant, based on https://github.com/liferay-frontend/liferay-portal/pull/4007#issuecomment-1956995423

**Comment by markocikos** [comment]
On hold, until https://github.com/brianchandotcom/liferay-portal/pull/147414 is merged, because of conflicts. I'll build my PR on top of that one.

File: modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts
Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts:81]: I reverted the `skip` commit, this was an issue on my end.

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:report:2544128693

**Comment by markocikos** [comment]
No unique failures, based on https://github.com/liferay-frontend/liferay-portal/pull/4007#issuecomment-1968528994

**Comment by markocikos** [comment]
ci:forward:force

**Pull request closed without merging**
